Cleanup ttnn.add uint16, uint32

mcw-anasuya opened the pull request:
### Ticket
#22330

### Problem description
Cleaning up uint16, uint32 ckernel implementations of add and include instr mod to template parameters

### What's changed
- Removed separate kernels of uint16, uint32 add in tt-metal and used LLK API
- Added instr modifier param to template
- Modified compute kernel to include uin16, uint32 tiles

LLK changes are tracked here: 
https://github.com/tenstorrent/tt-llk/pull/215 (Modify integer add ckernel to include instruction mode param)

### Checklist
- [x] Blackhole Testing
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15551238450) - PASSED
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15529294583) - same as Main
- [x] New/Existing tests provide coverage for changes

mcw-anasuya committed:
Cleanup uint16, uint32 add

Aswinmcw commented on tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_add_int.h:
```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
```
2025

Aswinmcw commented on tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_add_int.h:
```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
```
2025

Aswinmcw commented on tt_metal/third_party/tt_llk:
Rebase

VirdhatchaniKN reviewed (approved):
Please address the above comments

KalaivaniMCW commented on tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_add_int.h:
I don't think we need this file.. we can directly call _cal add_ from the llk_math file

KalaivaniMCW commented on tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_add_int.h:
I don't think we need this file.. we can directly call cal add from the llk_math file
pls refer bitwise PR

KalaivaniMCW commented on tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/llk_math_eltwise_binary_sfpu_add_int.h:
```diff
@@ -6,22 +6,25 @@
     uint dst_index0, uint32_t dst_index1, int vector_mode = VectorMode::RC) {
     llk_math_eltwise_binary_sfpu_params<APPROXIMATE>(
-        ckernel::sfpu::calculate_add_int32<APPROXIMATE, SIGN_MAGNITUDE_FORMAT>, dst_index0, dst_index1, vector_mode);
+        ckernel::sfpu::calculate_add_int<APPROXIMATE, SIGN_MAGNITUDE_FORMAT, INSTRUCTION_MODE>,
```
we can call _calculate_add_int_ fro here

KalaivaniMCW commented on tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_binary_sfpu_add_int.h:
```diff
@@ -6,22 +6,25 @@
 #include "llk_math_eltwise_binary_sfpu_init.h"
 #include "llk_math_eltwise_binary_sfpu_params.h"
-#include "ckernel_sfpu_add_int32.h"
+// #include "ckernel_sfpu_add_int.h"
```
to remove ?

KalaivaniMCW commented on tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/llk_math_eltwise_binary_sfpu_add_int.h:
```diff
@@ -6,22 +6,25 @@
 #include "llk_math_eltwise_binary_sfpu_init.h"
 #include "llk_math_eltwise_binary_sfpu_params.h"
-#include "ckernel_sfpu_add_int32.h"
+// #include "ckernel_sfpu_add_int.h"
```
remove ?

KalaivaniMCW reviewed (approved):
LGTM.

Aswinmcw reviewed (approved):
LGTM

mcw-anasuya commented:
@nvelickovicTT I've added the LLK commit in the description

mcw-anasuya commented:
@razorback3 @dongjin-na 
Please review this PR. I've updated the include path in [moreh_cumsum_sfpu_nc.cpp](https://github.com/tenstorrent/tt-metal/pull/22303/files#diff-bcf9ecc86118994be146e0aed1e9cff017730b3b7e46d94b5369b4ed4000b2bbL13) and renamed `add_int32_tile_init` to `add_int_tile_init`

fvranicTT commented:
> @mcw-anasuya this PR ingests quite a bit of LLK changes, thanks for linking the LLK PR for clarity of actual LLK changes needed here. @fvranicTT can we ingest LLK changes separately into tt-metal before this PR merges, so it actually ingests just the thing needed for this PR from LLK?

Yes, we need to take care of some changes that are causing failures in ttnn tests. I am aware of them, and I'm working on fixing this first. I'd like to uplift all of the other LLK changes before merging this one.

patrickroberts reviewed (approved):
Just want to confirm that the submodule update for `tt_metal/third_party/tt_llk` is an intentional part of this PR and not accidentally committed.

I ask because the submodule diff shows a lot more files changed than just 5 as the link to the corresponding PR in the description indicates.

mcw-anasuya commented:
@patrickroberts Those changes came up because I accidentally rebased the submodule separately. So I've linked the actual submodule changes in the description.
